ragna/source_storages/_chroma.py

```diff
@@ -96,19 +96,19 @@ def retrieve(
             include=["distances", "metadatas", "documents"],
         )
 
-        num_results = len(result["ids"])
+        num_results = len(result["ids"][0])
         result = {
             key: [None] * num_results if value is None else value[0]  # type: ignore[index]
             for key, value in result.items()
         }
         # dict of lists -> list of dicts
         results = [
-            {key: value[idx] for key, value in result.items()}
+            {key[:-1]: value[idx] for key, value in result.items()}
```

Comment:
Ignoring the last character here drops the "s" and thus makes the keys singular. Might sound nitpicky here, but I found myself confused when looking at the source again and having something like `id=result["ids"]` in the `Source` instantiation below.